src/Microsoft.AspNet.OData.Shared/DataModificationExceptionType.cs

```diff
@@ -0,0 +1,129 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNet.OData.Builder;
+
+namespace Org.OData.Core.V1
+{
+    /// <summary>
+    /// Represents a Message Type
+    /// </summary>
+    public class MessageType
+    {
+        /// <summary>
+        /// Code of message
+        /// </summary>
+        public string Code { get; set; }
+
+        /// <summary>
+        /// Actual message
+        /// </summary>
+        public string Message { get; set; }
+
+        /// <summary>
+        /// Severity of message
+        /// </summary>
+        public string Severity { get; set; }
+
+        /// <summary>
+        /// Target of message
+        /// </summary>
+        public string Target { get; set; }
+
+        /// <summary>
+        /// Details of message
+        /// </summary>
+        public string Details { get; set; }
+    }
+
+    /// <summary>
+    /// Represents an Exception Type
+    /// </summary>
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1012:AbstractTypesShouldNotHaveConstructors")]
+    public abstract class ExceptionType
+    {
+        /// <summary>
+        /// Instanciates a type of <see cref="ExceptionType"/>
+        /// </summary>
+        public ExceptionType()
+        {
+            ODataConventionModelBuilder builder = new ODataConventionModelBuilder();
+            builder.EntityType<DataModificationExceptionType>();
+      
+            builder.Namespace = typeof(DataModificationExceptionType).Namespace;
```

Comment:
You have a model builder here ..... :bangbang: 

What's it for? what's the usage of buider?